Eliminate Nvshmem get p2p

HermesCui opened the pull request:
Eliminated uccl::nvshmemi_get_p2p_ptr(dst_ptr, rank, dst_rank);

MaoZiming commented on gpu_driven/src/internode_ll.cu:
```diff
@@ -184,13 +184,18 @@ __global__ __launch_bounds__(1024, 1) void dispatch(
-          uccl::nvshmemi_ibgda_put_nbi_warp(dst_ptr, src_ptr, num_bytes_per_msg,
-                                            dst_rank, dst_expert_local_idx,
-                                            lane_id, slot_idx);
+#ifdef true
```
Feel free to delete these `#ifdef true` if they work!

MaoZiming commented on gpu_driven/src/internode_ll.cu:
```diff
@@ -292,12 +297,15 @@ __global__ __launch_bounds__(1024, 1) void dispatch(
+        reinterpret_cast<int*>(dst_offset), -num_tokens_sent - 1, dst_rank, sm_id,
+        dst_expert_local_idx, false, ring_addrs, num_ring_addrs);
     } else {
       st_release_sys_global(reinterpret_cast<int*>(dst_p2p_ptr),
```
Does `st_release_sys_global` still work for `dst_p2p_ptr`?

HermesCui replied:
It should not work it, but our current test does not trigger these {else}s.

MaoZiming replied:
Sounds good. We should have test that trigger intra-node transfers.

MaoZiming commented on gpu_driven/src/internode_ll.cu:
```diff
@@ -792,11 +800,14 @@ __global__ __launch_bounds__(1024, 1) void combine(
-      if (dst_p2p_ptr == 0) {
+#ifdef true
+      // auto dst_p2p_ptr = nvshmemi_get_p2p_ptr(dst_ptr, rank, dst_rank);
+      if (dst_rank != rank) {
```
if (`dst_rank != rank`), we might eventually need a way to tell (maybe from rank?) which ranks are on the same node.

MaoZiming replied:
Sounds good

HermesCui replied:
Oh, I remember that their test_intranode test calls intranode.cu, not internode_ll.cu. For intranode.cu, they assume that DeepEP will run on a single node with 8 GPUs.

HermesCui commented:
1. Replaced NVSHMEM dependency: Implemented `get_ipc_p2p_ptr() ` to determine intra-node vs inter-node communication without NVSHMEM
2. IPC handle exchange: Host creates and all-gathers IPC handles for GPU buffers, opening handles for intra-node GPU pairs
3. API extension: Added IPC pointer array parameter to support direct intra-node GPU memory access

MaoZiming commented on gpu_driven/include/ep_configs.cuh:
```diff
@@ -1,6 +1,6 @@
 #pragma once
 
-#define NUM_MAX_NVL_PEERS 8
+#define NUM_MAX_NVL_PEERS 1
```
I have added a new commit to set this dynamically.
